lib/collection/src/operations/point_ops.rs

```diff
@@ -216,6 +216,34 @@ pub enum PointInsertOperationsInternal {
     PointsList(Vec<PointStruct>),
 }
 
+impl PointInsertOperationsInternal {
+    pub fn point_ids(&self) -> Vec<ExtendedPointId> {
+        match self {
+            Self::PointsBatch(batch) => batch.ids.clone(),
+            Self::PointsList(points) => points.iter().map(|point| point.id).collect(),
+        }
+    }
+
+    pub fn remove_point_ids(&mut self, point_ids: &HashSet<ExtendedPointId>) {
+        match self {
+            Self::PointsBatch(batch) => {
+                for index in (0..point_ids.len()).rev() {
+                    if point_ids.contains(&batch.ids[index]) {
+                        batch.ids.swap_remove(index);
+                        batch.vectors.swap_remove(index);
```

Comment:
Operations must remain ordered, in case we update the same point ID twice in a single batch.